### PR TITLE
Oxidation file transactions

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

File: oxidation/crates/parsec_client_types/src/local_device.rs [hunk 1]
```diff
@@ -135,9 +135,9 @@ impl LocalDevice {
     // /// 1. Allowing for easier testing by patching this method in device-sepecific way
     // /// 2. Allowing for other implementation in the future allowing to track, check and
     // ///    possibly alter the production of timestamps.
```
Comment (FirelightFlagboy) [inline oxidation/crates/parsec_client_types/src/local_device.rs:137]: un-comment the doc-comment ;)

File: oxidation/crates/parsec_client_types/src/local_manifest.rs [hunk 2]
```diff
@@ -159,6 +159,65 @@ impl Chunk {
...
+impl LocalManifest {
+    pub fn id(&self) -> EntryID {
+        match self {
+            Self::File(manifest) => manifest.base.id,
+            Self::Folder(manifest) => manifest.base.id,
+            Self::Workspace(manifest) => manifest.base.id,
+            Self::User(manifest) => manifest.base.id,
+        }
+    }
+    pub fn need_sync(&self) -> bool {
```
Comment (FirelightFlagboy) [inline oxidation/crates/parsec_client_types/src/local_manifest.rs:179]:
```suggestion

    pub fn need_sync(&self) -> bool {
```
space

File: oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+
+        Ok(())
+    }
+
+    pub fn mark_prevent_sync_pattern_fully_applied(&self, pattern: &Regex) -> FSResult<()> {
+        // Mark the provided pattern as fully applied.
+
+        // This is meant to be called after one made sure that all the manifests in the
+        // workspace are compliant with the new pattern. The applied pattern is provided
+        // as an argument in order to avoid concurrency issues.
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs:235]: move the comment into doc-comment

File: oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+        })?;
+
+        Ok((re, fully_applied))
+    }
+
+    pub fn set_prevent_sync_pattern(&self, pattern: &Regex) -> FSResult<()> {
+        // Set the "prevent sync" pattern for the corresponding workspace
+
+        // This operation is idempotent,
+        // i.e it does not reset the `fully_applied` flag if the pattern hasn't changed.
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs:206]: move the comment into doc comment

File: oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+        for entry_id in keys {
+            self.ensure_manifest_persistent(entry_id)?;
+        }
+
+        Ok(())
+    }
+
+    // This method is not used in the code base but it is still tested
+    // as it might come handy in a cleanup routine later
+
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs:496]: into doc-comment

File: oxidation/libparsec/crates/core_fs/src/storage/mod.rs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+mod chunk_storage;
+mod local_database;
+mod manifest_storage;
+// Not used for the moment
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/core_fs/src/storage/mod.rs:6]:
```suggestion
/// Not used for the moment
```

File: oxidation/libparsec/crates/core_fs/src/workspacefs/mod.rs [hunk 7]
```diff
@@ -0,0 +1,5 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+// Not used for the moment
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/core_fs/src/workspacefs/mod.rs:3]:
```suggestion
/// Not used for the moment
```

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]
The header seems wrong, should be 2016-present

File: oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs (hunk 5 above)
Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs:496]: This comment seems to be for mainteners, not for users

File: oxidation/libparsec/crates/core_fs/src/storage/mod.rs (hunk 6 above)
Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/storage/mod.rs:6]: For mainteners, not users

File: oxidation/libparsec/crates/core_fs/src/workspacefs/mod.rs (hunk 7 above)
Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/workspacefs/mod.rs:3]: For mainteners, not users

File: oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs (hunk 5 above)
Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs:496]: ok

**Pull request closed without merging**